scene/gui/item_list.cpp

```diff
@@ -960,12 +960,30 @@ void ItemList::_notification(int p_what) {
 		Vector2 base_ofs = bg->get_offset();
 		base_ofs.y -= int(scroll_bar->get_value());
 
-		Rect2 clip(Point2(), size - bg->get_minimum_size() + Vector2(0, scroll_bar->get_value()));
+		Rect2 clip(-base_ofs, size);
+
+		int lo = 0;
```

Comment:
Maybe a quick comment explaining what the lo/hi search does would be useful? The previous code was relatively self explanatory as it just clips and draws stuff, but now it's a bit more involved :)

Reply:
Added comments and cleaned up the var scopes; oh C++11 lambdas, where are you?